### ISSUE TITLE
[interface-name-prefix] No error on interface naming without I

### ISSUE THREAD

**Issue by kushalmahajan** [typescript-eslint/typescript-eslint#943]
Below is my code in `.tsx` file extension.

```
interface AboutProps {
    someProp: string;
}
```
No error. Simple as that. Checked in *.d.ts file too. Same results

My configuration for `.eslintrc` file in CRA typescript cut out is 

```
{
    
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/eslint-recommended",
        "plugin:@typescript-eslint/recommended",
        "prettier",
        "prettier/@typescript-eslint",
        "plugin:@typescript-eslint/recommended-requiring-type-checking"
    ],
   
    "rules": {
        "@typescript-eslint/interface-name-prefix": [
            "error",
            {
                "prefixWithI": "always"
            }
        ]
    }
}
```

However, when I run CLI command for `eslint` it does reports the error of interfaces. What I need is to show the same in my editor too.

**Comment by bradzacher** [comment]
But you've configured `prefixWithI: "always"`? this tells the rule you always want to prefix with I....

Please have a look at the rule documentation for the correct configuration of the rule.

**Comment by kushalmahajan** [comment]
@bradzacher This was a small slip from my end to turn off lint plugin while doing some testing. Sorry!
